### ISSUE TITLE
Change in Microsoft Visual Basic PM Role

### ISSUE THREAD

**Issue by KathleenDollard** [dotnet/vblang#248]
This isn't really an issue, but I wanted to be sure the folks contributing here had heard this. 

@AnthonyDGreen left Microsoft :( 

See this for his thoughts: https://gist.github.com/AnthonyDGreen/d027ebb7052cce2400b670c7d4262014

I am taking over the VB PM role and looking forward to working with folks here.

**Comment by ocdtrekkie** [comment]
As a Chicagoan, welcome back @AnthonyDGreen.

**Comment by AdamSpeight2008** [comment]
*"The VB King is dead, long live the VB Queen!"*

**Comment by Bill-McC** [comment]
Congrats Kathleen.

**Comment by paul1956** [comment]
Welcome @KathleenDollard

**Comment by bandleader** [comment]
Welcome @KathleenDollard and good luck! @AnthonyDGreen, thank you for your wonderful work over the years, and we hope you'll remain part of the VB open-source community!

Kathleen, can you kindly give the OSS community an idea of what things will look like here on `vblang`? We have 194 issues open, and it seems to be a bit of a mess... perhaps only because Anthony was busy in preparation for his departure?

- there are tons of issues with no response from Microsoft at all 
- many issues are describing scenarios that should be combined into one feature _(must create issue for the full feature, describe which issues this supersedes. take in the most accepted syntax/ideas of the other issues, then CLOSE them)_
- some issues are not clearly defined, written in hard-to-understand English, or propose changes that cause grammar ambiguity (not everyone understands compilers) or solve too narrow a scope of scenarios, and the issue needs to be clarified, corrected, and sometimes integrated into a new issue.

Questions:

- Do you plan to review the current issues, get a feel for everything, and organize? Or are we heading toward a sort of "reset" and we should propose things to you again, maybe with a new tag, and reference the old issue?
- Will anyone else on the VB or Roslyn team who will be communicating with us?
- Following things here recently, I get the feeling that there are many issues describing well-defined, clearly valuable features -- even many that are simply "catch-up" features with the modern languages (C#/TypeScript/even ES6 embarrassingly enough!) that VB seems to be unfortunately "falling behind" -- not sure due to lack of interest at MS or what. / However, things just seem to drag on... there are only *two proposals* in the [proposals directory](https://github.com/dotnet/vblang/tree/master/proposals), and the [LDM meeting notes](https://github.com/dotnet/vblang/tree/master/meetings/2017) seem to be mostly discussing bugfixes, small updates, etc. / Please help us be realistic: If we open an issue here for a clearly-defined and valuable feature, *what are the chances that it will find its way to VB* within, say, 6 months?
- Until recently, the idea was that [VB and C# were first-class .NET languages, and new features were added to both](https://blogs.msdn.microsoft.com/scottwil/2010/03/09/vb-and-c-coevolution/). Last year, however, many of us longtime VB developers were disappointed when [Mads Torgerson implied](https://blogs.msdn.microsoft.com/dotnet/2017/02/01/the-net-language-strategy/) that this would be over. There was some apology about keeping VB beginner-friendly, but it seems that MS is not so interested in the language, probably mainly due to C#'s success. **If this is the case,** many of us here in the community here -- we are not the beginners -- would be soon grumbling and switching to C# for new projects (which many of us use as well anyway). However, we do have a preference for the beauty of the VB syntax, and if VB could go back to being not just beginner friendly **but also veteran-friendly,** adding features and keeping up with the modern world, we would love it. / Would you kindly be able to give a very transparent answer from MS as to the direction of VB?

We completely understand that it is not easy to head an OSS community, especially within the MS company culture, and we truly appreciate your efforts! Sorry to ask tough questions on your first days there -- it would just be really nice to know where we stand.

We look forward to your reply -- thank you!

**Comment by AnthonyDGreen** [comment]
Definitely planning to stay involved as a contributor here making suggestions, offering design feedback, implementing prototypes, etc. Kathleen and I have worked together for years while she was an MVP and I was the PM and I plan to keep working with her and the rest of the VB team now that I'm on the outside.

**Comment by AdamSpeight2008** [comment]
@KathleenDollard @AnthonyDGreen 
A wiki / use of the projects feature on this repo would be a nice addition, may also attract contributor to pick off outstanding tasks.
  * Proposals
     * Need specifications written,
  * Proof-Of-Concept prototypes.
    * References to forks, so others could contribute or help develop it.
  * Hints and Tips / Documentation.
    * Writing VB Analyser
    * Writing VB Code Fix
    * Parsing
    * Semantics
    * Binding
    * Lowering

**Comment by KathleenDollard** [comment]
My challenge has been deciding how to answer these questions in a way that contributed to repo organization. Because they strike at three distinct issues - me and the team, the repo, and community - I will answer them separately. I will talk about my perspective here, the [repo]() and [community](https://github.com/dotnet/vblang/issues/250) in separate issues. I am proposing in the repo issue that we create good separation into issues as I think that is important for context when people view our conversations later.

For those that don't know me, I'm a pragmatist. I believe passionately in bettering the experience of programmers as they sit down at the keyboard. While it was not my full time job until now, I have worked to better the programming experience across the last three decades. That passion has crossed languages, platforms and operating systems.

I have been involved with the Visual Basic and C# languages for about two decades (and trying to extend Clipper before that), so I am not unprepared. But I do not have the language skills, experience, and talent of of @madstorgersen and the amazing engineers. The languages group will work as a unit allowing Visual Basic to benefit. I also anticipate them continuing involvement here. All Roslyn languages (C#/VB) will continue to be considered for every change.

I see the challenges facing the Visual Basic community today to be larger than language, thus my [community issue](https://github.com/dotnet/vblang/issues/250).

The team and up the stack a ways has taken @anthonydgreen 's leaving as an opportunity to revisit the Visual Basic portion of the [.NET Language Strategy](https://blogs.msdn.microsoft.com/dotnet/2017/02/01/the-net-language-strategy/) released a year ago. I'm happy to say that we stand behind this strategy, although I also find it rather non-specific in it's terminology. I will not be able to answer all the questions right now because I want to take the time to get the right answers. I am committed to transparency, even when the answers aren't what we would rather hear.

On a personal note, I'd like to share with you that yesterday I sat on stage for the first time as the Visual Basic PM. It was a powerful experience to sit on stage between @madstorgersen and Anders Hejlsberg. It was also humbling to realize I different things to the role and have my own contributions to make.

**Comment by KathleenDollard** [comment]
@bandleader asked:

>Do you plan to review the current issues, get a feel for everything, and organize? Or are we heading toward a sort of "reset" and we should propose things to you again, maybe with a new tag, and reference the old issue?

I do not want a "reset." There is much value in this repo. I think review and organization will make it better.  

>Will anyone else on the VB or Roslyn team be communicating with us?

Yes, the team will still contribute in this repo. 

>Following things here recently, I get the feeling that there are many issues describing well-defined, clearly valuable features -- even many that are simply "catch-up" features with the modern languages (C#/TypeScript/even ES6 embarrassingly enough!) that VB seems to be unfortunately "falling behind" -- not sure due to lack of interest at MS or what. / However, things just seem to drag on... there are only two proposals in the proposals directory, and the LDM meeting notes seem to be mostly discussing bugfixes, small updates, etc. / Please help us be realistic: If we open an issue here for a clearly-defined and valuable feature, what are the chances that it will find its way to VB within, say, 6 months?

>Until recently, the idea was that VB and C# were first-class .NET languages, and new features were added to both. Last year, however,> many of us longtime VB developers were disappointed when Mads Torgerson implied that this would be over. There was some apology >about keeping VB beginner-friendly, but it seems that MS is not so interested in the language, probably mainly due to C#'s success. >If this is the case, many of us here in the community here -- we are not the beginners -- would be soon grumbling and switching to >C# for new projects (which many of us use as well anyway). However, we do have a preference for the beauty of the VB syntax, and if >VB could go back to being not just beginner friendly but also veteran-friendly, adding features and keeping up with the modern world, we would love it. / Would you kindly be able to give a very transparent answer from MS as to the direction of VB?

In the interest of organization, I would like you to move this to a new issue. The short answer is that the language strategy is still in place. But it is worth discussion because what does "first class citizen" mean? This is going to be an ongoing discussion, but I want it based on real pain points, not the current fancy fashion.

**Comment by reduckted** [comment]
> @KathleenDollard But it is worth discussion because what does "first class citizen" mean? 

Can I bring your attention to this relevant issue regarding "first class citizenship" 😄 
aspnet/Home#2738

**Comment by KathleenDollard** [comment]
Yes, and I'm trying to figure out the best way to manage a list of most significant pain points for VB developers. They aren't all language. 

ASP.NET templates are a particularly nuanced problem. If the templates go in the box, they have to be maintained forever, by a team that really doesn't get VB, and possibly with minimal real world testing. I think everyone might be better served if these templates were on a community site and Microsoft encouraged VB folks to look to that site for VB stuff. F# has such a site, although it is not where the templates are. (It's got it's own foundation, I hope VB could piggy back on .NET Foundation, but this is a proto-idea I haven't vetted yet.)

I'm going to let Eilon's answer stand there. At present ASP.NET has no plans to do this work. 

I wonder how weird it would be to have an issue here to discuss that
